File: Sample/SwiftToastObjcSample/SwiftToastObjcSample/RootViewController.m
//
//  RootViewController.m
//  SwiftToastObjcSample
//
//  Created by Fahim Farook on 10/27/14.
//  Copyright (c) 2014 Suyeol Jeon. All rights reserved.
//

#import "RootViewController.h"
#import "SwiftToastObjcSample-Swift.h"

@interface RootViewController ()

@end

@implementation RootViewController

-(IBAction)buttonTapped {
	[[Toast makeToast:@"Basic SwiftToast"] show];
	// Second toast - you normally don't have to do this, this is just for chaining toasts with a delay
	dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 2.5 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
		[[Toast makeToast:@"You can set a pre-set duration or a duration value in seconds." duration:ToastConfig.durationNormal] show];
	});
	// Third toast
	dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 6.5 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
		[[Toast makeToast:@"With delay, SwiftToast will be shown after delay." duration:5 delay:1] show];
	});
}

@end

File: Sample/SwiftToastSwiftSample/SwiftToastSwiftSample/RootViewController.swift
/*
 * RootViewController.swift
 *
 * Original code by Su Yeol Jeon 2013-2014.
 * Branched and modified by Fahim Farook 2014 to add additional functionality which did
 * not meet with original author's requirements/specifications
 *
 */

import UIKit

class RootViewController: UIViewController {
	
	@IBAction func showToast1() {
		Toast.makeToast("Basic SwiftToast", position:ToastConfig.positionDefault, offset:48).show()
	}
	
	@IBAction func showToast2() {
		Toast.makeToast("You can set a pre-set duration or a duration value in seconds.", duration:ToastConfig.durationNormal, position:ToastConfig.positionTop, offset:66).show()
	}
	
	@IBAction func showToast3() {
		Toast.makeToast("With delay, SwiftToast will be shown after a delay.", delay:1, duration:5, position:ToastConfig.positionCenter, offset:ToastConfig.offsetDefault).show()
	}
}
